larpandoracontent/LArObjects/LArThreeDSlidingConeFitResult.cc

```diff
@@ -117,16 +117,24 @@ void ThreeDSlidingConeFitResult::GetSimpleConeList(const unsigned int nLayersFor
     const TrackStateMap &trackStateMap(this->GetTrackStateMap());
     const unsigned int nLayers(trackStateMap.size());
 
-    if (nLayers < nLayersForConeFit)
+    if (nLayers + 1 < nLayersForConeFit + nCones)
         throw StatusCodeException(STATUS_CODE_INVALID_PARAMETER);
 
-    unsigned int nConeSamplingSteps(0);
-    const unsigned int coneInterval((nCones > 1) ? (nLayers - nLayersForConeFit) / (nCones - 1) : std::numeric_limits<unsigned int>::max());
+    // Calculate intervals and offsets such that cones are evenly spaced along sliding fit and are equidistant from the extremal layers
+    const unsigned int coneInterval((nCones > 1) ? (nLayers - nLayersForConeFit) / (nCones - 1) : 1);
+    const ThreeDSlidingFitResult &slidingFitResult(this->GetSlidingFitResult());
+    const CartesianVector coneDisplacement(slidingFitResult.GetGlobalMaxLayerPosition() - slidingFitResult.GetGlobalMinLayerPosition());
+    const bool isForward(coneDisplacement.GetZ() > std::numeric_limits<float>::epsilon());
+    const unsigned int coneOffset1((nLayers - nLayersForConeFit - (nCones - 1) * coneInterval) / 2);
+    const unsigned int coneOffset2((1 == nLayers % 2) && isForward ? 1 : 0);
+    const unsigned int coneOffset(coneOffset1 + coneOffset2);
```

Comment:
This logic is pretty tortuous, but seems sound - will have to read through it again later to double check!